internal/service/ec2/ec2_fleet.go

```diff
@@ -658,7 +665,11 @@ func resourceFleetRead(ctx context.Context, d *schema.ResourceData, meta interfa
 	}.String()
 	d.Set("arn", arn)
 	d.Set("context", fleet.Context)
-	d.Set("excess_capacity_termination_policy", fleet.ExcessCapacityTerminationPolicy)
```

Comment:
Can you please explain why you are using conditional logic before setting the value in state? As per best practice, we should write the state if a value is supplied, we do not normally wrap any conditional logic around state management.

Reply:
The value is only applicable to fleet type `maintain`